packages/@atjson/source-html/src/converter/social-embeds.ts

```diff
@@ -0,0 +1,124 @@
+import Document, { Annotation } from "@atjson/document";
+import OffsetSource, { IframeEmbed } from "@atjson/offset-annotations";
+import UrlSource from "@atjson/source-url";
+
+function isInstagramOrTwitterBlockquote(annotation: Annotation<any>) {
+  if (annotation.type !== "blockquote") {
+    return false;
+  }
+
+  let classes = annotation.attributes.class;
+
+  if (!classes) {
+    return false;
+  }
+
+  let classList = classes.split(" ");
+  return (
+    classList.includes("instagram-media") || classList.includes("twitter-tweet")
+  );
+}
+
+function isFacebookDiv(annotation: Annotation<any>) {
+  if (annotation.type !== "div") {
+    return false;
+  }
+
+  return (
+    annotation.attributes.dataset &&
+    annotation.attributes.dataset.href &&
+    new URL(annotation.attributes.dataset.href).host === "www.facebook.com"
+  );
+}
+
+export default function(doc: Document) {
+  doc.where({ type: "-html-iframe" }).update(iframe => {
+    let { height, width, src: url } = iframe.attributes;
+    let urlDoc = UrlSource.fromRaw(url).convertTo(OffsetSource);
+    let embed;
+    if (urlDoc.annotations.length) {
+      embed = urlDoc.annotations[0];
+      embed.start = iframe.start;
+      embed.end = iframe.end;
+    } else {
+      embed = new IframeEmbed({
+        start: iframe.start,
+        end: iframe.end,
+        attributes: {
+          url
+        }
+      });
+    }
+    embed.attributes = { ...embed.attributes, height, width };
+    doc.replaceAnnotation(iframe, embed);
+  });
+
+  /**
+   * Instagram/Twitter embeds in blockquotes:
+   *   <blockquote class="instagram-media" data-instgrm-permalink="url">
+   *     ...
+   *     <a href="https://www.instagram.com/p/{id}">
```

Comment:
Instagram has the ability for people to include / exclude the caption; we aren't going to support this _right now_, but it would be lovely to be able to add this in the future.

![Screen Shot 2019-11-05 at 12 40 51](https://user-images.githubusercontent.com/177701/68231696-88320680-ffc9-11e9-85f8-44484990ffbc.png)